flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobClientTest.java

```diff
@@ -487,4 +487,11 @@ private static void uploadJarFile(
 			validateGetAndClose(blobClient.getInternal(jobId, blobKeys.get(0)), testFile);
 		}
 	}
+
+	@Test
+	public void testUnresolvedInetSocketAddress() throws Exception {
+		try (BlobClient client = new BlobClient(
+			InetSocketAddress.createUnresolved("localhost", getBlobServer().getPort()), getBlobClientConfig())) {
+		}
```

Comment:
Could you extend this test, so that it checks that it actually connects to the `BlobServer`?

Reply:
I'm adding a new method to BlobClient to check if the socket is connected and then adding an assert here.